<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BamSliceRWorkFlow • bamSliceR.Bioc2024.Demo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BamSliceRWorkFlow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bamSliceR.Bioc2024.Demo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/How_to_generate_txs_coordinates_gencode_gff.html">BamSliceRWorkFlow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/huangyizR/bamSliceR-Bioc2024-Demo" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BamSliceRWorkFlow</h1>
                        <h4 data-toc-skip class="author">Peter
Huang</h4>
            <address class="author_afil">
      Van Andel Institute/Michigan State
University<br><h4 data-toc-skip class="date">16 July 2024</h4>
      

      <div class="hidden name"><code>How_to_generate_txs_coordinates_gencode_gff.rmd</code></div>

    </address>
</div>

    
    
<div class="section level2">
<h2 id="abstract">Abstract<a class="anchor" aria-label="anchor" href="#abstract"></a>
</h2>
<p>analysis. We will use the cell line mixture data from Tian <em>et
al.</em> <span class="citation">[@tianComprehensiveCharacterizationSinglecell2021]</span>.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>lower throughput, to accommodate this, we profile a smaller
sub-sample of cells in our <em>FLT-seq</em> protocol<span class="citation">[@tianComprehensiveCharacterizationSinglecell2021]</span>:
<img src="FLT-seq.png" width="800"></p>
<p>There are also many other protocols that combine single-cell with
long-read sequencing, including <em>scRaCH-seq</em> which profiles gene
panel <span class="citation">[@pengSinglecellRapidCapture2024]</span>,
<em>LR-Split-seq</em> which uses combinatorial barcoding and does not
require microfluidic equipments <span class="citation">[@rebboahMappingModelingGenomic2021]</span>.</p>
</div>
<div class="section level2">
<h2 id="set-up">Set-up<a class="anchor" aria-label="anchor" href="#set-up"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind, colnames,</span></span>
<span><span class="co">##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,</span></span>
<span><span class="co">##     intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste,</span></span>
<span><span class="co">##     pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames,</span></span>
<span><span class="co">##     sapply, setdiff, table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/libPaths.html" class="external-link">.libPaths</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/tmp/Rtmpq64xyU/temp_libpath3877bebcad" "/usr/local/lib/R/site-library"         </span></span>
<span><span class="co">## [3] "/usr/local/lib/R/library"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="flames-pipeline">
<code>FLAMES</code> pipeline<a class="anchor" aria-label="anchor" href="#flames-pipeline"></a>
</h2>
<div class="section level3">
<h3 id="read-pre-processing">Read pre-processing<a class="anchor" aria-label="anchor" href="#read-pre-processing"></a>
</h3>
</div>
<div class="section level3">
<h3 id="alignment">Alignment<a class="anchor" aria-label="anchor" href="#alignment"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="data-pre-processing">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing"></a>
</h2>
</div>
<div class="section level2">
<h2 id="mutation-analysis">Mutation analysis<a class="anchor" aria-label="anchor" href="#mutation-analysis"></a>
</h2>
<div class="section level3">
<h3 id="mutation-discovery">Mutation discovery<a class="anchor" aria-label="anchor" href="#mutation-discovery"></a>
</h3>
</div>
<div class="section level3">
<h3 id="mutation-calling">Mutation calling<a class="anchor" aria-label="anchor" href="#mutation-calling"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="differential-transcript-experssion-analysis">Differential Transcript Experssion analysis<a class="anchor" aria-label="anchor" href="#differential-transcript-experssion-analysis"></a>
</h2>
</div>
<div class="section level2">
<h2 id="remove-lowly-expressed-isoforms">remove lowly expressed isoforms<a class="anchor" aria-label="anchor" href="#remove-lowly-expressed-isoforms"></a>
</h2>
</div>
<div class="section level2">
<h2 id="visualize-the-pseudo-bulk-samples">visualize the pseudo-bulk samples<a class="anchor" aria-label="anchor" href="#visualize-the-pseudo-bulk-samples"></a>
</h2>
</div>
<div class="section level2">
<h2 id="differential-transcript-usage-analysis">Differential Transcript Usage analysis<a class="anchor" aria-label="anchor" href="#differential-transcript-usage-analysis"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] rtracklayer_1.65.0   GenomicRanges_1.57.1 GenomeInfoDb_1.41.1  IRanges_2.39.2      </span></span>
<span><span class="co">## [5] S4Vectors_0.43.2     BiocGenerics_0.51.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.9                  SparseArray_1.5.22          bitops_1.0-7               </span></span>
<span><span class="co">##  [4] lattice_0.22-6              digest_0.6.36               grid_4.4.1                 </span></span>
<span><span class="co">##  [7] evaluate_0.24.0             fastmap_1.2.0               jsonlite_1.8.8             </span></span>
<span><span class="co">## [10] Matrix_1.7-0                restfulr_0.0.15             httr_1.4.7                 </span></span>
<span><span class="co">## [13] UCSC.utils_1.1.0            XML_3.99-0.17               Biostrings_2.73.1          </span></span>
<span><span class="co">## [16] codetools_0.2-20            textshaping_0.4.0           jquerylib_0.1.4            </span></span>
<span><span class="co">## [19] abind_1.4-5                 cli_3.6.3                   rlang_1.1.4                </span></span>
<span><span class="co">## [22] crayon_1.5.3                XVector_0.45.0              Biobase_2.65.0             </span></span>
<span><span class="co">## [25] DelayedArray_0.31.9         cachem_1.1.0                yaml_2.3.9                 </span></span>
<span><span class="co">## [28] S4Arrays_1.5.4              tools_4.4.1                 parallel_4.4.1             </span></span>
<span><span class="co">## [31] BiocParallel_1.39.0         GenomeInfoDbData_1.2.12     Rsamtools_2.21.0           </span></span>
<span><span class="co">## [34] SummarizedExperiment_1.35.1 curl_5.2.1                  R6_2.5.1                   </span></span>
<span><span class="co">## [37] BiocIO_1.15.0               matrixStats_1.3.0           lifecycle_1.0.4            </span></span>
<span><span class="co">## [40] zlibbioc_1.51.1             fs_1.6.4                    htmlwidgets_1.6.4          </span></span>
<span><span class="co">## [43] ragg_1.3.2                  desc_1.4.3                  pkgdown_2.1.0              </span></span>
<span><span class="co">## [46] bslib_0.7.0                 systemfonts_1.1.0           xfun_0.46                  </span></span>
<span><span class="co">## [49] GenomicAlignments_1.41.0    MatrixGenerics_1.17.0       knitr_1.48                 </span></span>
<span><span class="co">## [52] rjson_0.2.21                htmltools_0.5.8.1           rmarkdown_2.27             </span></span>
<span><span class="co">## [55] compiler_4.4.1              RCurl_1.98-1.16</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>fig.path<span class="op">=</span><span class="st">'Figs/gff.parse/'</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="subset-of-gencode-v36-annotation">Subset of GENCODE.v36.annotation<a class="anchor" aria-label="anchor" href="#subset-of-gencode-v36-annotation"></a>
</h4>
<p>Transcriptome BAM files lack the information of genomic coordinates
when mapping reads to reference transcripts sequences. Convention GFF3
files, from GENCODE for example, contains the gene annotation on the
reference chromosomes, which are not compatible with Transcriptome BAM.
In order to easily annotate the tallied variants from transcriptome BAM,
we provide get_txs_coords_of_gff() to calculate the coordinates of
transcripts for each features entity (except “gene” feature), based on
the given genomic coordinates. The function would maintain the internal
structure of the gff3 file (for example, the hierarchy of the features
for each genes.)</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#coords.column.names = c("seqid", "type", "start", "end", "gene_name")</span></span>
<span></span>
<span><span class="co">#gencode.v36.file = system.file("extdata", "gencode.v36.annotation.subset.gff3", </span></span>
<span><span class="co">#                             package = "bamSliceR")</span></span>
<span><span class="co">#gencode.v36.df = readGFF(gencode.v36.file)</span></span>
<span><span class="co">#gencode.v36.df[,coords.column.names]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="generate-gencode-v36-transcripts-annotation">Generate gencode.v36.transcripts.annotation<a class="anchor" aria-label="anchor" href="#generate-gencode-v36-transcripts-annotation"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#genomic.column.names = c("g_seqid", "g_start", "g_end")</span></span>
<span><span class="co">#get_txs_coords_of_gff( gencode.file = gencode.v36.file, isSaveGenomicCoords = TRUE) -&gt; gencode.v36.transcripts.df</span></span>
<span><span class="co">#gencode.v36.transcripts.df[,c(coords.column.names, genomic.column.names)]</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Peter Huang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
